### DISCUSSION TITLE
Maintenance tracker

### DISCUSSION THREAD

**Discussion by cwanja** [teslamate-org/teslamate#245] [Ideas]
**Describe the solution you'd like**
In addition to the "Geofence" page, a new "Maintenance" page that has extremely limited (and simple) options:

- Mileage of maintenance performed (numbers only)
- Work performed (free text, allow for multi-line format [return character])

**Describe alternatives you've considered**
Excel or other spreadsheet; retaining workorders from Tesla in email

**Additional context**
If you want to get fancy, you could include a section that has the "recommended maintenance" from Tesla based upon the car type (S, 3, X, etc.) that is already collected. Happy to collect all of this for you from the manuals. Would need to include a "last updated" such that it can be compared against the manual on Tesla's webpage.

**Comment by mat1990dj** [comment]
Teslafi has this, and I'm currently tracking cabin filter changes, tire rotations, and tire replacements

**Comment by aaronspruit** [comment]
I was just going to add a new idea for this. 

I'd love to have a feature where teslamate notifies me when I should be doing maintenance based on typical intervals.  I'd love to have this notification just be something on the main page, something not too dissimilar to the 'notification' there's a new version available.

I imagine not everyone would want this feature and so it'd be great to have the option to enable/disable it in the settings and that'd be a good place to put the "last maintenance done" user inputs.

https://www.tesla.com/support/vehicle-maintenance

No idea if this is of interest, but for things such as air filters, you could embed a referral link too.

**Comment by zipizigi** [comment]
I agree with this idea.

After entering the time, location, and ODO, the user simply enters a description.
Of course, teslamate can know location and ODO through time. (Autocomplete and editable)

In the Description field, enter the user's desired description.
An example might be:
- Tire Replacement
- Exchange tire position
- Accident occurrence and repair
- Replace washer fluid
- Filter replacement
- Various repairs
- car wash
- Purpose of visiting the place


The information entered can be viewed through a new dashboard.
Or can show it in your existing timeline dashboard.

Users can now log a little more information about what they did and when.
This feature can be particularly helpful for items that require periodic maintenance. For example, tires and filters.

**Comment by steiny2k** [comment]
I use [Grafana annotations on visualizations](https://grafana.com/docs/grafana/latest/dashboards/build-dashboards/annotate-visualizations/) for this. In the drive after the maintenance (or any other event) I put an annotation into the drive timeline panel with a certain tag (i.e. "Maintenance"). It's a free-text field you can describe what work has been done.

And I created a dashboard showing all the [annotations as a list](https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/annotations/), which can be filtered by the tags. It's quite straightforward...

**Comment by IKyriazis** [comment]
This would be a great addition to enable reminders to rotate tires, get wheel alignments, change air filter, etc. at a custom mile threshold or date threshold. Are we considering adding this feature?

I'd especially like to be emailed every time one of these events gets triggered.

That way I can ensure even tire wear and get the most length out of my tires and cash in on the warranty if the tire doesn't last as long as it should.

**Reply by IKyriazis** [reply]
this is actually really funny and coincidental but tesla just released an update where you can track the maintenance required a few days after i wrote this post

**Reply by kvandivo** [reply]
Where did Tesla release this?  I just looked through my app options and don't see anything.  (Android, Model S 2017)

**Reply by cwanja** [reply]
In vehicle.
